docs/reference/profiles.md

```diff
@@ -0,0 +1,155 @@
+
+- `runtimeClassName` : See https://kubernetes.io/docs/concepts/containers/runtime-class/ for a description and links to any additional documentation about Pod Runtime Class
+- `tolerations` : https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/ for a description and links to any additional documentation about Tolerations.
+- `affinity` : https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity for a description and links to any additional documentation about Node Affinity.
+- `podSecurityContext` : https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for a description and links to any additional documentation about the Pod Security Context.
+
+The configuration use the exact options that you find in the Kubernetes documentation.
+
+### Examples
+
+#### Use an Alternative RuntimeClass
+A popular alternative container runtime class is [gVisor](https://gvisor.dev/) that provides additional sandboxing between containers. If you have created a cluster that is using gVisor, you will need to set the `runTimeClass` on the Pods that are created. This is not exposed in the OpenFaaS API, but it can be set via a Profile.
+
+1. Install the latest `faas-netes` release and the CRD. The is most easily done with [`arkade`](https://github.com/alexellis/arkade)
+    ```sh
+    arkade install openfaas
+    ```
+    This default installation will enable Profiles.
+2. Create a Profile to apply the runtime class
+    ```yaml
+    kubectl apply -f- << EOF
+    kind: Profile
+    apiVersion: openfaas.com/v1
+    metadata:
+        name: gvisor
+        namespace: openfaas
+    spec:
+        runtimeClassName: gvisor
+    EOF
+    ```
+3. Let your developers know that they need to use this annotation
+    ```
+    com.openfaas.profile: gvisor
+    ```
+
+The following stack file will deploy a SHA512 generating file in a cluster with gVisor
+
+```yaml
+provider:
+  name: openfaas
+  gateway: http://127.0.0.1:8080
+
+functions:
+  stronghash:
+    skip_build: true
+    image: functions/alpine:latest
+    fprocess: "sha512sum"
+    annotations:
+      com.openfaas.profile: gvisor
+```
+
+#### Use Tolerations and Affinity to Separate Workloads
+The OpenFaaS API already exposes the Kubernetes `NodeSelector` via [`constraints`](/docs/reference/yaml#function-constraints). This provides a very simple selection based on labels on Nodes. The affinity/anti-affinity and taint/tolerations features significantly expand the types of constraints you can express. This allows you to more accurately keep certain workloads together on the same nodes or oppositely on separate loads. E.g. keeping CPU intensive loads on more expensive nodes with more CPU versus keeping moderate or low CPU workloads on cheaper nodes with a less CPU.
```

Comment:
"already exposes" <- this is accurate, but I wonder how well it will age, in 1-2 years?

Could you mention that it's an alternative approach?